docs/concepts/runtime_capabilities.md

```diff
@@ -103,21 +103,59 @@ Both the [`elastic` connector](../reference/connectors/elastic.md) and the [`htt
+
+### Example 1: kafka source & udp sink
+
+**Source**: The kafka source supports guaranteed delivery. A failure on a message after the latest acknowledged offset will revert to that message and replay all data since then.
+
+**Sink**: The udp sink has no delivery guarantee any message reaching it will be automatically delivered
+
+**Result**: All messages read from Kafa will be automatically acknowledged.
+
+### Example 2: udp source & kafka sink
+
+**Source**: Upd as transport has no guarantees, the upstream system sending the messages does not know if it arrives or is handled.
+
+**Sink**: Depending on its setting, Kafka acknowledges successful storage of the message and will fail an event if it can't store it.
+
+**Result**: The upstream system has no guarantees its messages reach somewhere. If a message is delivered to tremor it will make a best effort attempt to deliver it, but if Kafka fails to persist the event, a replay can't happen due to UDP's limitation.
+
+### Example 3: http source & kafka sink
+
+**Source**: The WebSocket source will acknowledge successful delivery of the message to its upstream system but does not support replays or other methods of dealing with failures.
+
+**Sink**: Depending on its setting, Kafka acknowledges successful storage of the message and will fail an event if it can't store it.
+
+**Result**: Messages that are delivered to tremor will be delivered to Kafka. If Kafka fails to persist the message, an error is sent back to the upstream system as a response to the HTTP request. How the upstream system handles this failure is up to the service.
+
+### Example 4: http source & wal connector & kafka sink
+
+**Source**: The WebSocket source will acknowledge successful delivery of the message to its upstream system but does not in itself support replays or other methods of dealing with failures.
+
+**Intermidiary**: This setup uses a wal connector as an intermediary, effectively splitting the system into two pipelines. One pipelein being http to wal the other being wal to kafka.
+
+**Sink**: Depending on its setting, Kafka acknowledges successful storage of the message, and will fail an event if it can't store it.
+
+**Result 1 (http -> wal)**: The wal connector, as long as it doesn't run full, will always acknowledge messages it stores, so all data is successfully sent to the HTTP endpoint, no matter the state of the downstream system, will be acknowledged. This frees the HTTP API form from having to consider retries.
+
+**Result 2 (wal -> kafka)**: The wal connector acts pretty similar to Kafka when looking at its source nature. It will replay all messages since a failed event that wasn't acknowledged before. So failure in Kafka for storing messages will lead to a re-transmission from the wal - this happens in a transparent matter from the point of view of the HTTP connector.
```

Comment:
```suggestion
**Result 2 (wal -> kafka)**: The [`wal` connector](../connectors/wal.md) acts pretty similar to the `kafka_consumer`. Whenever an event is marked as failed, it will replay all events since that one. So failure in the `kafka_producer` producing events will lead to a re-transmission from the `wal` - this happens in a transparent matter from the point of view of the `http_server` connector.
```